torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -60,11 +60,34 @@
 if TYPE_CHECKING:
     from collections import OrderedDict  # noqa: F401
 
+_TORCHDISTX_AVAIL = True
+try:
+    from torchdistx import fake, deferred_init
+except ImportError:
+    _TORCHDISTX_AVAIL = False
+
 
 FSDP_WRAPPED_MODULE = "_fsdp_wrapped_module"
 FSDP_PREFIX = FSDP_WRAPPED_MODULE + "." + FPW_MODULE + "."
 
 
+def _default_meta_device_init_fn(module):
+    """
+    Default initializer for modules initialized on the meta device.
+    """
+    # TODO: move module to device_ids here once device_id is available.
+    module.to_empty(device=torch.cuda.current_device())
+    try:
```

Comment:
do we want to enforce the same seed among ranks here? possible it is not needed after broadcasting PR is landed, otherwise it requires the same seed for ranks

Reply:
Discussed offline - maybe not since application and not library should explicitly seed. It will also be fixed out of the box with parameter broadcast PR.